internal/rbd/controllerserver.go

```diff
@@ -1207,14 +1211,12 @@ func (cs *ControllerServer) doSnapshotClone(
 		}
 	}()
 
-	if parentVol.isEncrypted() {
-		cryptErr := parentVol.copyEncryptionConfig(&cloneRbd.rbdImage, false)
-		if cryptErr != nil {
-			log.WarningLog(ctx, "failed copy encryption "+
-				"config for %q: %v", cloneRbd, cryptErr)
+	cryptErr := parentVol.copyEncryptionConfig(&cloneRbd.rbdImage, false)
+	if cryptErr != nil {
+		log.WarningLog(ctx, "failed copy encryption "+
```

Comment:
use err here we dont need a new error and correct the return err and log as an error as we are returning error from here?

Reply:
> use err here we dont need a new error and correct the return err and log as an error as we are returning error from here?

Using err will cause a cleanup due to defer func at new line no. 1206.
I don't think we need to cleanup snapshot and clone incase of encryptionconfig copy fails, 
we can pick up from the same spot in a follow-up request
and copy the encryption config here https://github.com/ceph/ceph-csi/blob/43fe5c34cd12546e99657b28e73930d2ba12682a/internal/rbd/controllerserver.go#L1107.

I'll log as log.Error and return cryptErr at line 1219.

Does this sound good @Madhu-1 ?

Reply:
Sounds good but If the encryption copy fails why to keep snapshot and clone?

Reply:
> Sounds good but If the encryption copy fails why to keep snapshot and clone?

yes, :+1:  after some more thinking it makes sense to return err and cleanup everything. 

If cryptErr is intermittent, we should be fine with continuing the process in a follow-up request
But if is repetitive due to wrong configuration or something, the stale snap and clone can never be
deleted automatically since createSnapshot Req can never be successfully completed.